h2o-persist-s3/src/main/java/water/persist/H2OCredentialProviderFactory.java

```diff
@@ -0,0 +1,76 @@
+package water.persist;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.alias.CredentialProvider;
+import org.apache.hadoop.security.alias.CredentialProviderFactory;
+import water.util.Log;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+
+public class H2OCredentialProviderFactory extends CredentialProviderFactory {
+
+    @Override
+    public CredentialProvider createProvider(URI uri, Configuration configuration) {
+        if ("hex".equals(uri.getScheme()) && getClass().getName().equals(uri.getHost()))
+            return new H2OCredentialProvider(new PersistS3.H2OAWSCredentialsProviderChain());
+        else
+            return null;
```

Comment:
Shouldn't an exception be thrown when it's gets to this state? (Or an error logged at least?)

Reply:
This was intentional - it leaves room for adding additional "h2o" providers. If no provider for "hex://" can be found - the exception will be thrown from the Hadoop code.

Reply:
ok, thanks for your explanation!